tc/core/polyhedral/codegen_llvm.cc

```diff
@@ -625,12 +627,21 @@ std::unique_ptr<llvm::Module> emitLLVMKernel(
   cg.halide_cg.get_module()->setDataLayout(dataLayout);
   cg.halide_cg.get_module()->setTargetTriple(
       llvm::EngineBuilder().selectTarget()->getTargetTriple().str());
-  cg.createSignature(
+  auto entry = cg.createSignature(
       scop.halide.inputs,
       scop.halide.outputs,
       scop.halide.params,
       specializedName);
-  cg.CodeGen(islCg.astNode);
+  auto exit = cg.emitAst(islCg.astNode, entry);
+  cg.halide_cg.get_builder().SetInsertPoint(exit);
+  cg.halide_cg.get_builder().CreateRetVoid();
+
+  if (llvm::verifyModule(*cg.halide_cg.get_module())) {
+    LOG(ERROR) << cg.str();
+    llvm::verifyModule(*cg.halide_cg.get_module(), &llvm::outs());
+    throw std::runtime_error("LLVM generated module is invalid.");
```

Comment:
Why do we throw a `runtime_error` instead of using `TC_CHECK(false)`.
I understand the check throws the same type of exception, but it is not guaranteed.

Reply:
legacy, will update